### PR TITLE
deprecation fixes for project, file-details & analysis sections

### PR DESCRIPTION
**Pull request by future-pirate-king**

### CODE REVIEW

File: app/initializers/irene.js [hunk 1]
```diff
@@ -3,137 +3,188 @@ import installPendo from 'irene/utils/install-pendo';
...
+   * @param {string} envKey - The key of the environment variable
+   * @returns {string} The value of the environment variable
+   */
+  getEnv(envKey) {
+    const hostKey = 'IRENE_API_HOST';
+    this.assertEnvKey(envKey);
+
+    if (this.isAvailableInRuntimeENV(envKey)) {
+      const runtimeValue = this.getRuntimeObject()[envKey];
+      if (envKey === hostKey && runtimeValue === '/') {
```
Comment (Yibaebi) [inline app/initializers/irene.js:36]: Add space above.

File: app/initializers/irene.js [hunk 2]
```diff
@@ -3,137 +3,188 @@ import installPendo from 'irene/utils/install-pendo';
...
+        return '';
       }
       return runtimeValue;
```
Comment (Yibaebi) [inline app/initializers/irene.js:39]: Add space above and other return statements below.

File: app/models/user.ts [hunk 3]
```diff
@@ -32,16 +32,16 @@ export default class UserModel extends Model {
   @attr('string')
   declare lastName: string;
 
-  @hasMany('project', { inverse: 'owner' })
+  @hasMany('project', { inverse: 'owner', async: true })
   declare ownedProjects: AsyncHasMany<ProjectModel>;
 
-  @hasMany('project')
+  @hasMany('project', { async: true, inverse: null })
   declare projects: AsyncHasMany<ProjectModel>;
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/models/user.ts:39]: @future-pirate-king Can you review this as well?

File: config/environment.js [hunk 4]
```diff
@@ -198,7 +198,7 @@ module.exports = function (environment) {
     },
     rootURL: '/',
     favicon: '/images/favicon.ico',
-    locationType: 'auto',
+    locationType: 'history',
```
Comment (Yibaebi) [inline config/environment.js:201]: Would this change not affect browsers that do not support the HTML5 History API?

**Review by Yibaebi — CHANGES_REQUESTED** [review]

File: app/initializers/irene.js (hunk 1 above)
Comment (future-pirate-king) [reply app/initializers/irene.js:36]: done

File: app/initializers/irene.js (hunk 2 above)
Comment (future-pirate-king) [reply app/initializers/irene.js:39]: done

File: config/environment.js (hunk 4 above)
Comment (future-pirate-king) [reply config/environment.js:201]: No `auto` also defaults to `history` thats why `auto` option is getting removed in ember 5

File: app/models/user.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (future-pirate-king) [reply app/models/user.ts:39]: `projects` in user model is not related any field in project model

**Review by Yibaebi — APPROVED** [review]

**Pull request merged**